# Aviation Risk Analysis Project

## Introduction
The  project  analyzes  this [Aviation dataset](https://www.kaggle.com/datasets/prathamsharma123/aviation-accidents-and-incidents-ntsb-faa-waas?resource=download&select=airline_accidents.csv) to find which aircrafts are safest for a new aviation business.
The goal is to help the  management make data-driven decisions on the fleet selection and operations.

## Objectives
1. Clean and prepare the dataset
2. Explore and visualize accident patterns
3. Reccommend the safest aircraft and practices.


##  Exploratory Data Analysis

In [2]:
#Importing the necessary libraries
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [ ]:
#Loading the dataset into the notebook

df = pd.read_csv('./Aviation_Data.csv')

#Display the top 5 rows
df.head()

c:\Users\DELL\anaconda3\envs\learn-env\lib\site-packages\IPython\core\interactiveshell.py:3145: DtypeWarning: Columns (6,7,28) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country Latitude Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1   BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2    Saltville, VA  United States  36.9222  -81.8781          NaN   
3       EUREKA, CA  United States      NaN       NaN          NaN   
4       Canton, OH  United States      NaN       NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [ ]:
#To check the shape 
df.shape  #The dataset has 90348 rows and 31 columns

(90348, 31)

In [4]:
#Looking at the basic info of the coulmns of the dataframe
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50249 non-null  object 
 9   Airport.Name            52790 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87572 non-null  object 
 14  Make                    88826 non-null

In [8]:
#looking at the summary of the statistical measures
df.describe() 

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

In [ ]:
#Looking at all the columns to maybe identify some relevant columns
df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

In [22]:
#Looking at some of the columns that could be relevant

df.loc[:,['Aircraft.Category','Make','Model','Injury.Severity','Weather.Condition','Total.Uninjured','Total.Minor.Injuries','Total.Serious.Injuries','Total.Fatal.Injuries']].head(10)

Aircraft.Category               Make         Model Injury.Severity  \
0               NaN            Stinson         108-3        Fatal(2)   
1               NaN              Piper      PA24-180        Fatal(4)   
2               NaN             Cessna          172M        Fatal(3)   
3               NaN           Rockwell           112        Fatal(2)   
4               NaN             Cessna           501        Fatal(1)   
5          Airplane  Mcdonnell Douglas           DC9       Non-Fatal   
6               NaN             Cessna           180        Fatal(4)   
7          Airplane             Cessna           140       Non-Fatal   
8          Airplane             Cessna          401B       Non-Fatal   
9               NaN     North American  NAVION L-17B       Non-Fatal   

  Weather.Condition  Total.Uninjured  Total.Minor.Injuries  \
0               UNK              0.0                   0.0   
1               UNK              0.0                   0.0   
2               IMC              NaN                   NaN   
3               IMC              0.0                   0.0   
4               VMC              0.0                   NaN   
5               VMC             44.0                   1.0   
6               IMC              0.0                   0.0   
7               VMC              2.0                   0.0   
8               IMC              2.0                   0.0   
9               IMC              0.0                   3.0   

   Total.Serious.Injuries  Total.Fatal.Injuries  
0                     0.0                   2.0  
1                     0.0                   4.0  
2                     NaN                   3.0  
3                     0.0                   2.0  
4                     2.0                   1.0  
5                     NaN                   NaN  
6                     0.0                   4.0  
7                     0.0                   0.0  
8                     0.0                   0.0  
9                     0.0                   0.0

In [23]:
#Looking at some of the columns that could be relevant

df.loc[:,['Aircraft.Category','Make','Model','Injury.Severity','Weather.Condition','Total.Uninjured','Total.Minor.Injuries','Total.Serious.Injuries','Total.Fatal.Injuries']].tail(10)

Aircraft.Category                         Make      Model  \
90338               NaN                       AIRBUS   A330-243   
90339               NaN                        PIPER       PA28   
90340               NaN                       CESSNA       172F   
90341               NaN  GRUMMAN AMERICAN AVN. CORP.      AA-5B   
90342               NaN                  AIR TRACTOR      AT502   
90343               NaN                        PIPER  PA-28-151   
90344               NaN                     BELLANCA       7ECA   
90345          Airplane   AMERICAN CHAMPION AIRCRAFT      8GCBC   
90346               NaN                       CESSNA       210N   
90347               NaN                        PIPER  PA-24-260   

      Injury.Severity Weather.Condition  Total.Uninjured  \
90338             NaN               NaN              0.0   
90339       Non-Fatal               NaN              3.0   
90340           Minor               NaN              0.0   
90341           Minor               NaN              1.0   
90342           Fatal               NaN              0.0   
90343           Minor               NaN              0.0   
90344             NaN               NaN              0.0   
90345       Non-Fatal               VMC              1.0   
90346             NaN               NaN              0.0   
90347           Minor               NaN              1.0   

       Total.Minor.Injuries  Total.Serious.Injuries  Total.Fatal.Injuries  
90338                   0.0                     0.0                   0.0  
90339                   0.0                     0.0                   0.0  
90340                   0.0                     1.0                   0.0  
90341                   0.0                     1.0                   0.0  
90342                   0.0                     0.0                   1.0  
90343                   0.0                     1.0                   0.0  
90344                   0.0                     0.0                   0.0  
90345                   0.0                     0.0                   0.0  
90346                   0.0                     0.0                   0.0  
90347                   0.0                     1.0                   0.0

## Quick Data Summary

- **Rows / columns:** The dataset contains **90348 rows** and **31 columns**.
- **Data types:** Most columns are `object` (text); there are 5 numeric columns (`float64`) for injury counts and engine number.
- **Missingness:** Several fields have substantial missing values (for example `Latitude`/`Longitude` , `Aircraft.Category` and `FAR.Description`).  
- **Key columns that could be relevant :** `Aircraft Ctaegory`,`Make`, `Model`, `Injury.Severity`, `Total.Fatal.Injuries`, `Total.Serious.Injuries`, `Weather.Condition` could be relevant to our aviation analysis.
- **Initial concerns / notes:**  
  - Some location fields (`Latitude`, `Longitude`) are strings and many values are missing — may be unusable unless cleaned.  
  - `Event.Date` is currently an object (string) and should be parsed to datetime for trend analysis.  
  - `Make` and `Model` appear to have good coverage but may contain inconsistent capitalization or spelling which will need standardization.


## Data Cleaning

### Dealing with missing values 

In [25]:
#First ceheck for duplicates

df.duplicated().sum()

1390

In [26]:
#Dropping the duplicates
df = df.drop_duplicates()
df.duplicated().sum()

0

In [ ]:
#Checking the no. of rows and columns after dropping duplicates
df.shape

(88958, 31)

In [29]:
#Checking the counts of the missing values in the columns in descsnding order

df.isnull().sum().sort_values(ascending=False)


Schedule                  76376
Air.carrier               72310
FAR.Description           56935
Aircraft.Category         56671
Longitude                 54585
Latitude                  54576
Airport.Code              38709
Airport.Name              36168
Broad.phase.of.flight     27234
Publication.Date          15299
Total.Serious.Injuries    12579
Total.Minor.Injuries      12002
Total.Fatal.Injuries      11470
Engine.Type                7146
Report.Status              6450
Purpose.of.flight          6261
Number.of.Engines          6153
Total.Uninjured            5981
Weather.Condition          4561
Aircraft.damage            3263
Registration.Number        1386
Injury.Severity            1069
Country                     295
Amateur.Built               171
Model                       161
Make                        132
Location                    121
Event.Date                   69
Accident.Number              69
Event.Id                     69
Investigation.Type            0
dtype: i

In [37]:
#To check the percentage of missingness against the length of the dataframe for each column  to now have a threshold of whether to drop or replace 
percentage_of_missingness = round(df.isna().sum() *100 / len(df) , 2)
percentage_of_missingness.sort_values(ascending = False)

Schedule                  85.86
Air.carrier               81.29
FAR.Description           64.00
Aircraft.Category         63.71
Longitude                 61.36
Latitude                  61.35
Airport.Code              43.51
Airport.Name              40.66
Broad.phase.of.flight     30.61
Publication.Date          17.20
Total.Serious.Injuries    14.14
Total.Minor.Injuries      13.49
Total.Fatal.Injuries      12.89
Engine.Type                8.03
Report.Status              7.25
Purpose.of.flight          7.04
Number.of.Engines          6.92
Total.Uninjured            6.72
Weather.Condition          5.13
Aircraft.damage            3.67
Registration.Number        1.56
Injury.Severity            1.20
Country                    0.33
Amateur.Built              0.19
Model                      0.18
Make                       0.15
Location                   0.14
Event.Date                 0.08
Accident.Number            0.08
Event.Id                   0.08
Investigation.Type         0.00
dtype: f

## Explanation of the above
Some columns have a lot of missing values, especially:

 - Schedule (85.86%)

 - Air.carrier (81.29%)

 - FAR.Description (64.00%)

 - Aircraft.Category (63.71%)

 - Latitude and Longitude (61% each)

That means those columns are mostly empty and may not add much value for my analysis  since my objectives are about accident patterns and safety insights so i can opt to drop them

In [42]:
# Drop columns with too many missing values 

cols_to_drop = ['Schedule', 'Air.carrier', 'FAR.Description','Aircraft.Category', 'Latitude', 'Longitude']

df = df.drop(columns=cols_to_drop)

# Check again to confirm they were removed
df.info()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 88958 entries, 0 to 90347
Data columns (total 25 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88958 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Airport.Code            50249 non-null  object 
 7   Airport.Name            52790 non-null  object 
 8   Injury.Severity         87889 non-null  object 
 9   Aircraft.damage         85695 non-null  object 
 10  Registration.Number     87572 non-null  object 
 11  Make                    88826 non-null  object 
 12  Model                   88797 non-null  object 
 13  Amateur.Built           88787 non-null  object 
 14  Number.of.Engines       82805 non-null

In [43]:
# Check remaining missing values after dropping columns
df.isna().sum().sort_values(ascending=False)


Airport.Code              38709
Airport.Name              36168
Broad.phase.of.flight     27234
Publication.Date          15299
Total.Serious.Injuries    12579
Total.Minor.Injuries      12002
Total.Fatal.Injuries      11470
Engine.Type                7146
Report.Status              6450
Purpose.of.flight          6261
Number.of.Engines          6153
Total.Uninjured            5981
Weather.Condition          4561
Aircraft.damage            3263
Registration.Number        1386
Injury.Severity            1069
Country                     295
Amateur.Built               171
Model                       161
Make                        132
Location                    121
Event.Date                   69
Accident.Number              69
Event.Id                     69
Investigation.Type            0
dtype: int64